internal/controller/authenticator.go

```diff
@@ -181,11 +194,34 @@ func SetUpOAuthConnection(
 	}
 	tokenSource = GetTokenSource(ctx, oauthConfig, tlsConfig)
 	creds := credentials.NewTLS(tlsConfig)
+
+	proxyDialer := func(ctx context.Context, addr string) (net.Conn, error) {
+		proxyFunc := configureProxy(logger)
+		if proxyFunc == nil {
+			return net.Dial("tcp", addr)
+		}
+		proxyURL, err := proxyFunc(&http.Request{URL: &url.URL{Host: addr}})
+		if err != nil {
+			logger.Warn("Invalid HTTPS proxy configured; ignoring proxy settings", zap.Error(err))
+			return net.Dial("tcp", addr)
+		}
+		if proxyURL == nil { // No proxy configured
+			return net.Dial("tcp", addr)
+		}
+		conn, err := proxy.Dial(ctx, "tcp", addr)
+		if err != nil {
+			logger.Error("Failed to connect via proxy; falling back to direct connection", zap.Error(err))
```

Comment:
Info -> Warn to be consistent with the log above?
```suggestion
			logger.Warn("Failed to connect via proxy; falling back to direct connection", zap.Error(err))
```